test/DGmethods/sphere/advection_sphere_ssp33.jl

```diff
@@ -0,0 +1,384 @@
+            starttime[] = now()
+        else
+            @info @sprintf("""Update
+                           simtime = %.16e
+                           runtime = %s
+                           Δmass   = %.16e""",
+                           ODESolvers.gettime(TimeIntegrator),
+                           Dates.format(convert(Dates.DateTime,
+                                                Dates.now()-starttime[]),
+                                        Dates.dateformat"HH:MM:SS"),
+            abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe))
+        end
+        nothing
+    end
+
+    # Set up the information callback
+    cbmass = GenericCallbacks.EveryXSimulationSteps(1000) do
+        @info @sprintf """Update
+            Δmass   = %.16e""" abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe)
+    end
+
+    step = [0]
+    mkpath("vtk")
+    cbvtk = GenericCallbacks.EveryXSimulationSteps(1000) do (init=false)
+        outprefix = @sprintf("vtk/advection_sphere_%dD_mpirank%04d_step%04d",
+                             3, MPI.Comm_rank(mpicomm), step[1])
+        @debug "doing VTK output" outprefix
+        writevtk(outprefix, Q, spacedisc, ("ρ", ))
+        step[1] += 1
+        nothing
+    end
+    #------------Set Callback Info--------------------------------#
+
+    # Perform Time-Integration
+    solve!(Q, TimeIntegrator; timeend=timeend, callbacks=(cbinfo, cbmass, cbvtk))
+
+    # Print some end of the simulation information
+    if integration_testing
+        error = euclidean_distance(Q, Qe) / norm(Qe)
+        Δmass = abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe)
+        @info @sprintf """Finished
+            error = %.16e
+            Δmass = %.16e
+            """ error Δmass
+    else
+        error = euclidean_distance(Q, Qe) / norm(Qe)
+        Δmass = abs(weightedsum(Q) - weightedsum(Qe)) / weightedsum(Qe)
+        @info @sprintf """Finished
+            error = %.16e
+            Δmass = %.16e
+            """ error Δmass
+    end
+
+    # return diagnostics
+    return (error, Δmass)
+end
+#}}} Main
+
+#{{{ Run Script
+function run(mpicomm, Nhorizontal, Nvertical, N, timeend, DFloat, dt, ti_method, ArrayType)
+    Rrange=range(DFloat(1); length=Nvertical+1, stop=2)
+    topl = StackedCubedSphereTopology(mpicomm,Nhorizontal,Rrange; boundary=(1,1))
+    (error, Δmass) = main(mpicomm, DFloat, topl, N, timeend, ArrayType, dt, ti_method)
+end
+#}}} Run Script
+
+#{{{ Run Program
+using Test
+let
+    MPI.Initialized() || MPI.Init()
+    Sys.iswindows() || (isinteractive() && MPI.finalize_atexit())
+    mpicomm=MPI.COMM_WORLD
+
+    ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+    loglevel = ll == "DEBUG" ? Logging.Debug :
+    ll == "WARN"  ? Logging.Warn  :
+    ll == "ERROR" ? Logging.Error : Logging.Info
+    logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+    global_logger(ConsoleLogger(logger_stream, loglevel))
+    @static if Base.find_package("CUDAnative") !== nothing
```

Comment:
if #246 lands first this should be
```suggestion
@static if haspkg("CUDAnative")
 ```